File: springboot_libraryapi/src/main/java/libraryapi/models/dataTransferObjects/UpdateBook.java
package libraryapi.models.dataTransferObjects;

import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

public record UpdateBook(@Pattern(regexp="^(978-?|979-?)?\\d{1,5}-?\\d{1,7}-?\\d{1,6}-?\\d{1,3}$", message="isbn is not in the correct format.") String isbn, @Size(min=1, max=300, message="title must have between {min} and {max} characters.") String title, @Size(min=1, max=100, message="author must have between {min} and {max} characters.") String author) {}

File: gin_libraryapi-main/models/datatransferobjects/updatebook.go
package datatransferobjects

type UpdateBook struct {
	Isbn   string `json:"isbn" binding:"omitempty,isbn"`
	Title  string `json:"title" binding:"omitempty,min=0,max=300"`
	Author string `json:"author" binding:"omitempty,min=0,max=300"`
}
